plugins/spotify/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/spotify/PlayerLogic.java

```diff
@@ -346,142 +257,60 @@ public boolean back() {
 	 * @return a hashMap with the keys name and artist
 	 */
 	public Map<String, String> getCurrentSong() {
-		HashMap<String, String> result = new HashMap<>();
-		if (checkPlayerState()) {
-			GetInformationAboutUsersCurrentPlaybackRequest getInformationAboutUsersCurrentPlaybackRequest = this.auth
-					.getSpotifyApi().getInformationAboutUsersCurrentPlayback().build();
-			try {
-				CurrentlyPlayingContext currentlyPlayingContext = getInformationAboutUsersCurrentPlaybackRequest
-						.execute();
-				result.put("name", currentlyPlayingContext.getItem().getName());
-				String artists = "";
-				for (ArtistSimplified artist : currentlyPlayingContext.getItem().getArtists()) {
-					artists = artists.concat(artist.getName()).concat(" ");
-
-				}
-				result.put("artist", artists);
-				return result;
-			} catch (SpotifyWebApiException | IOException e) {
-				this.logger.error(e.getMessage());
-				return new HashMap<>();
-			}
+		CurrentlyPlayingContext currentlyPlayingContext = spotifyAPICalls.getCurrentSong();
+		if (currentlyPlayingContext != null) {
+			Track[] track = { currentlyPlayingContext.getItem() };
+			return search
+					.createTrackOutput(new Paging.Builder<Track>().setItems(track).build(), SpotifyConstants.TYPE_TRACK)
+					.get(0);
 		}
 		return new HashMap<>();
+
 	}
 
 	/**
 	 * this method controls the volume of the player
 	 * 
 	 * @param volumeString
 	 *            allowed strings: mute, max, up, down
-	 * @return a int from 0-100. This represent the Volume in percent. if the
-	 *         Playerstate incorrect return -1
+	 * @return a int from 0-100. This represent the Volume in percent. if the volume
+	 *         is unknown the return value is -1
 	 */
 	public int setVolume(String volumeString) {
-		if (checkPlayerState()) {
+		int volume = spotifyAPICalls.getVolume();
+		if (volume != -1) {
 			switch (volumeString) {
 			case "mute":
-				setVolume(0);
-				this.volume = 0;
-				break;
+				spotifyAPICalls.setVolume(VOLUME_MUTE_VALUE);
+				return VOLUME_MUTE_VALUE;
 			case "max":
-				setVolume(100);
-				break;
+				spotifyAPICalls.setVolume(VOLUME_MAX_VALUE);
+				return VOLUME_MAX_VALUE;
 			case "up":
-				if (this.volume + 10 <= 100) {
-					this.volume += 10;
-					setVolume(this.volume);
-
+				if (volume + VOLUME_UPDOWN_VALUE <= VOLUME_MAX_VALUE) {
```

Comment:
Maybe if the volume's at 95% I want to increase it to 100%? 
I think removing the `if` and using `MIN(MAX_VOLUME, v + up_step)` would be better